pandas_log/pandas_execution_stats.py

```diff
@@ -114,32 +141,44 @@ def get_logs_for_specifc_method(self):
     def _repr_html_(self):
         pass
 
-    def __repr__(self):
+    def __repr__(self, verbose, copy_ok):
         # Step title
         func_sig = get_signature_repr(
-            self.fn, self.fn_args, self.full_signature
+            self.cls, self.fn, self.fn_args, self.full_signature
         )
         step_number = (
             "X"
-            if self.fn.__name__ in PANDAS_ADDITIONAL_METHODS_TO_OVERIDE
+            if self.fn.__name__ in DATAFRAME_ADDITIONAL_METHODS_TO_OVERIDE
             else self.execution_stats.step_number
         )
         step_title = f"{step_number}) {func_sig}"
 
         # Step Metadata stats
-        logs, tips = self.get_logs_for_specifc_method()
-        metadata_stats = f"\033[4mMetadata\033[0m:\n{logs}" if logs else ""
-        metadata_tips = f"\033[4mTips\033[0m:\n{tips}" if tips else ""
-
-        # Step Execution stats
-        exec_time_humanize = (
-            f"* Execution time: Step Took {self.execution_stats.exec_time}."
-        )
-        exec_input_memory_humanize = f"* Input Dataframe size is {self.execution_stats.input_memory_size}."
-        exec_output_memory_humanize = f"* Output Dataframe size is {self.execution_stats.output_memory_size}."
-        execution_stats = f"\033[4mExecution Stats\033[0m:\n\t{exec_time_humanize}\n\t{exec_input_memory_humanize}\n\t{exec_output_memory_humanize}"
+        logs, tips = self.get_logs_for_specifc_method(verbose, copy_ok)
+        if not logs:
```

Comment:
If logs is an empty string (ie the method isn't patched and verbose=False so we don't want to print the default logging message), the else block here will still cause it to print some output. So eg if logs and tips are both empty strings the output is "\n{step_title}\n\t\n\t{execution_stats}\n\t". I was operating under the principle that if the method isn't patched and verbose=False then nothing should be printed, but if you'd prefer it to still log just the execution stats in that case I can clean up the return value so that it does that without extra newlines and tabs. And yes either way it is checking if logs/tips are empty strings twice which should be fixed.

Reply:
I think that its cleaner to use silent as indication whether a log should be printed to stdout or not.
I use verbose for methods which in most usecases are useless for example whether the dataframe been copied or not.

I am however open if you think for suggestion

Reply:
No that makes sense. I cleaned up the output and added it back in for methods which don't have a patched method.